scalding-macros/src/test/scala/com/twitter/scalding/macros/MacroOrderingProperties.scala

```diff
@@ -0,0 +1,540 @@
+/*
+Copyright 2012 Twitter, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package com.twitter.scalding.macros
+
+import org.scalatest.{ FunSuite, ShouldMatchers }
+import org.scalatest.prop.Checkers
+import org.scalatest.prop.PropertyChecks
+import scala.language.experimental.macros
+import com.twitter.scalding.serialization.{ OrderedSerialization, Law, Law1, Law2, Law3, Serialization }
+import java.nio.ByteBuffer
+import org.scalacheck.Arbitrary.{ arbitrary => arb }
+import java.io.{ ByteArrayOutputStream, InputStream }
+
+import com.twitter.bijection.Bufferable
+import org.scalacheck.{ Arbitrary, Gen, Prop }
+import com.twitter.scalding.serialization.JavaStreamEnrichments
+
+import scala.collection.immutable.Queue
+
+trait LowerPriorityImplicit {
+  implicit def primitiveOrderedBufferSupplier[T] = macro com.twitter.scalding.macros.impl.OrderedSerializationProviderImpl[T]
+}
+
+object LawTester {
+  def apply[T: Arbitrary](laws: Iterable[Law[T]]): Prop =
+    apply(implicitly[Arbitrary[T]].arbitrary, laws)
+
+  def apply[T](g: Gen[T], laws: Iterable[Law[T]]): Prop =
+    laws.foldLeft(true: Prop) {
+      case (soFar, Law1(name, fn)) => soFar && Prop.forAll(g)(fn).label(name)
+      case (soFar, Law2(name, fn)) => soFar && Prop.forAll(g, g)(fn).label(name)
+      case (soFar, Law3(name, fn)) => soFar && Prop.forAll(g, g, g)(fn).label(name)
+    }
+}
+
+object ByteBufferArb {
+  implicit def arbitraryTestTypes: Arbitrary[ByteBuffer] = Arbitrary {
+    for {
+      aBinary <- Gen.alphaStr.map(s => ByteBuffer.wrap(s.getBytes("UTF-8")))
```

Comment:
for my own understanding, this should be the same as `s.getBytes("ASCII")` right?